numpy/core/tests/test_matrix_transpose.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+from hypothesis import given
+from hypothesis.extra.numpy import array_shapes
+
+import numpy as np
+from numpy.testing import assert_array_equal
+
+
+def test_matrix_transpose_equals_swapaxes_1d():
+    arr = np.arange(48)
+    assert_array_equal(arr.T, arr.mT)
+
+
+def test_matrix_transpose_equals_swapaxes_2d():
+    arr = np.arange(48).reshape((6, 8))
+    assert_array_equal(arr.T, arr.mT)
+
+
+@given(shape=array_shapes(min_dims=3))
+def test_matrix_transpose_equals_swapaxes(shape):
+    num_of_axes = len(shape)
+    total_elements = np.prod(shape)
+    arr = np.arange(total_elements).reshape(shape)
```

Comment:
This is pretty naive. It could create a very large array, that may cause problems depending on which shapes are generated.

Reply:
Use `broadcast_to`?  Trusts the content is correct, but that seems OK (at least beyond 1-2 explicit tests).  TBH, hypothesis seems a bit overkill in this case, its not like there is any weird complexity we need to cover beyond high dimensions.

Reply:
Done.